pypesto/petab/importer.py

```diff
@@ -726,24 +738,27 @@ def create_problem(
 
         # Raise error if the correct calculator is not used.
 
-        if self._non_quantitative_data_types:
+        if self._non_quantitative_data_types is not None and any(
+            data_type in self._non_quantitative_data_types
+            for data_type in [ORDINAL, CENSORED, SEMIQUANTITATIVE]
+        ):
             if not isinstance(objective.calculator, InnerCalculatorCollector):
                 raise AssertionError(
-                    f"If there are ordinal, censored or nonlinear-monotone measurements, the `calculator` attribute of the `objective` has to be {InnerCalculatorCollector} and not {objective.calculator}."
+                    f"If there are ordinal, censored or semi-quantitative measurements, the `calculator` attribute of the `objective` has to be {InnerCalculatorCollector} and not {objective.calculator}."
```

Comment:
semiquantitative vs semi-quantitative

Reply:
True, also in other places.

Reply:
Will change to `semiquantitative` everywhere.